src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -107,6 +107,25 @@ export const CollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
         description: "Bid count on lots open for bidding",
         deprecationReason: "Use nested field in `auction` instead",
       },
+      curatorsPick: {
+        type: GraphQLBoolean,
+        description:
+          "Artwork is part of either the Curators' Pick Emerging or Blue Chip collections",
+        resolve: async (artwork, {}, ctx) => {
+          const CURATED_COLLECTION_SLUGS = [
+            "curators-picks-blue-chip-artists",
+            "curators-picks-emerging-artists",
+          ]
+
+          const checks = await Promise.all(
+            CURATED_COLLECTION_SLUGS.map(async (slug) => {
+              const collection = await ctx.marketingCollectionLoader(slug)
+              return collection.artwork_ids.includes(artwork._id)
+            })
+          )
+          return checks.includes(true)
+        },
+      },
```

Comment:
I believe for booleans, we don't need to do a non-null since it's only `true` or `false`, but I could be wrong 🤷

Reply:
Reading from some [schema design](https://www.apollographql.com/docs/technotes/TN0023-nullability/) reference about nullability, it seems to recommend (or at least at the beginning of introducing a new field) fields to be nullable when it involves external services. It also makes iteration of the schema easier (i.e. changing from nullable to non-nullable is not a breaking change, but is the opposite way).

Reply:
Yes - I've generally advocated this as well. Errors from upstream can happen, and since we don't do the `OrError` pattern for everything - you need to consider the upstream error.

Since an upstream error typically results in a `null` (unless you `catch` yourself) - that indeed would imply to make a field where the resolver fetches data nullable. Making it non-null will blow up on an error in a [too] noisy way.

You _can_ make it non-null and return a default in a `catch`, but then you're also likely swallowing the error - and the thing could be broken w/o anyone knowing.

_Not_ swallowing the error and keeping a field nullable means that the error should bubble up - whether or not anyone actually looks at it is debatable (😆 ) - but at least it's not explicitly being silenced via `catch { return false }` or something like that.

_Generally_ speaking (and of course there are always exceptions) - fields w/ resolvers that fetch data from upstream should likely be nullable. Any field where the resolver is doing stuff with _already_ fetched data - that can definitely be non-null if desired.

In general, nullable -> non-null is not a breaking change as @starsirius points out, so for iteration/at first - nullable makes sense for that reason too.